packages/common/src/ui/components/inputs/CurrencyInput/CurrencyInput.tsx

```diff
@@ -24,49 +25,18 @@ const StyledCurrencyInput = styled(RCInput)(({ theme }) => ({
   },
 }));
 
-// TODO: I think the implementation of this could be swapped out by making a
-// headless component that uses something like currency.js in a custom hook
-// to manage the currency value. Then you could use whatever input component
-// you liked. That would just take a little more time than I have right now!
 export const CurrencyInput: FC<CurrencyInputProps> = ({
   allowNegativeValue = false,
-  prefix = '$',
-  decimalSeparator = '.',
-  decimalsLimit = 2,
   allowDecimals = true,
-  decimalScale = 2,
-  value = 0,
+  defaultValue = 0,
   onChangeNumber,
   maxWidth,
   ...restOfProps
 }) => {
-  // Buffer the internal input value so we can account for the decimal separator
-  // which, when input will result in the input being NaN (e.g. '42.') when cast
-  // to a number. In this case, we want to keep the value as '42.' in the input
-  // but not persisted. Though, whenever I have an uncontrolled component I
-  // generally will always regret it, so sync the passed value up with the buffer.
-  const [buffer, setBuffer] = React.useState<string | undefined>(String(value));
-  useEffect(() => {
-    if (value !== Number(buffer)) {
-      setBuffer(String(value));
-    }
-  }, [value]);
+  const { c, options, language } = useCurrency();
```

Comment:
I think we're preferring not to use `intl` in `ui`, but I didn't see another way of encapsulating this in the component. I really didn't think any consumers of the component should really have to handle all this stuff